### PR TITLE
Backport improvements made for objects panel

### PR DESCRIPTION
**Pull request by 4ian**
- Add properties to old objects.
  - Allow some properties to specify that changing them should re-render everything, this will be useful for the particles emitter.
- Fixed some existing compact components not to grow outside of their parent (min-width: 0)
- Added CompactToggleField, CompactResourceSelectorWithThumbnail, CompactTextAreaField, CompactColorField.
  - ResourceThumbnail reworked to allow the size to be specified.
  - ColorPicker reworked to allow the size to be specified. 

TODO:
- [x] Toggle button should read its color from the theme.

### CODE REVIEW

File: newIDE/app/src/UI/CompactTextAreaField/CompactTextField.module.css [hunk 1]
```diff
@@ -0,0 +1,85 @@
+.container {
```
Comment (ClementPasteau) [inline newIDE/app/src/UI/CompactTextAreaField/CompactTextField.module.css:1]: rename this file to CompactTextAreaField

Comment (4ian) [reply newIDE/app/src/UI/CompactTextAreaField/CompactTextField.module.css]: Yes

File: newIDE/app/src/ResourcesList/CompactResourceSelectorWithThumbnail/index.js [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+          className={classNames({
+            [classes.compactResourceSelector]: true,
+          })}
+        >
+          <input type="text" value={resourceName} />
+        </div>
+      </div>
+      <ElementWithMenu
+        element={
+          <IconButton size="small" onClick={() => {}}>
```
[the remark this answers is not in the document]
Comment (4ian) [reply newIDE/app/src/ResourcesList/CompactResourceSelectorWithThumbnail/index.js:257]: Indeed or it should be set by ElementWithMenu

File: newIDE/app/src/CompactPropertiesEditor/index.js [hunk 3]
```diff
@@ -360,41 +378,32 @@ const CompactPropertiesEditor = ({
   );
 
   const renderInputField = React.useCallback(
-    (field: ValueField) => {
+    (field: PrimitiveValueField) => {
       if (field.name === 'PLEASE_ALSO_SHOW_EDIT_BUTTON_THANKS') return null; // This special property was used in GDevelop 4 IDE to ask for a Edit button to be shown, ignore it.
```
[the remark this answers is not in the document]
Comment (4ian) [reply newIDE/app/src/CompactPropertiesEditor/index.js:382]: We'll remove this later (we need to remove it in Sprite and Physics behavior - and make this an attribute of ObjectMetadata or BehaviorMetadata, like `setDisplayButtonForFullEditor(true)`)

File: Core/GDCore/Project/Effect.h [hunk 4]
```diff
@@ -32,6 +32,9 @@ class GD_CORE_API Effect {
   }
   const gd::String& GetEffectType() const { return effectType; }
 
+  void SetFolded(bool fold = true) { folded = fold; }
+  bool IsFolded() const { return folded; }
```
[the remark this answers is not in the document]
Comment (4ian) [reply Core/GDCore/Project/Effect.h:36]: Good catch, it's not. We could add it.

[the remark this answers is not in the document]
Comment (ClementPasteau) [reply Core/GDCore/Project/Effect.h:36]: I don't think this is something we want actually, I think it's good not to save this.

**Pull request merged**